src/state-of-the-layer/state-of-the-layer.js

```diff
@@ -0,0 +1,290 @@
+/**
+ * Ceros Plugin for preserving an Experience's state via its URL
+ * @version 0.1.0
+ * @support support@ceros.com
+ *
+ * This plugin allows people using the Ceros Studio's SDK pallet to tag components
+ * to enable them to both control, and be controlled by, parameters in
+ * the Experience's URL. It can be used to remember user's actions
+ * between pages or to trigger actions like showing layers or
+ * jumping to anchor points when the Experience is opened
+ * by the user.
+ *
+ * This plugin is licensed under the MIT license. A copy of this license and
+ * the accompanying source code is available at https://github.com/ceros/ceros-plugins
+ */
+(function() {
+    "use strict";
+
+    /**
+     * Class to mange state represented in the URL
+     *
+     * @constructor
+     */
+    var UrlHashManager = function() {
+
+        var hash = window.location.hash;
+
+        // The State represented in the URL, and the order its keys were added
+        this.state = {};
+        this.keyOrder = [];
```

Comment:
why do we need two different collections? why dont we have an array of objects or something that way they are already in order